apps/frontend/src/components/fap/Proposals/AssignProposalsToFaps.tsx

```diff
@@ -0,0 +1,161 @@
+import Alert from '@mui/material/Alert';
+import Button from '@mui/material/Button';
+import Container from '@mui/material/Container';
+import Grid from '@mui/material/Grid';
+import InputAdornment from '@mui/material/InputAdornment';
+import Typography from '@mui/material/Typography';
+import { Form, Formik } from 'formik';
+import React, { useContext } from 'react';
+
+import FormikUIAutocomplete from 'components/common/FormikUIAutocomplete';
+import { UserContext } from 'context/UserContextProvider';
+import {
+  FapInstrument,
+  FapInstrumentInput,
+  ProposalViewInstrument,
+  UserRole,
+} from 'generated/sdk';
+import { useFapsData } from 'hooks/fap/useFapsData';
+import { getUniqueArrayBy } from 'utils/helperFunctions';
+
+type AssignProposalToFapsProps = {
+  close: () => void;
+  assignProposalsToFaps: (
+    fapInstruments: FapInstrumentInput[]
+  ) => Promise<void>;
+  proposalInstruments: (ProposalViewInstrument[] | null)[];
+  proposalFapInstruments?: (FapInstrument | null)[];
+};
+
+const AssignProposalsToFaps = ({
+  close,
+  assignProposalsToFaps,
+  proposalInstruments,
+  proposalFapInstruments,
+}: AssignProposalToFapsProps) => {
+  const { currentRole } = useContext(UserContext);
+  const { faps, loadingFaps } = useFapsData({
+    filter: '',
+    active: true,
+    role: currentRole as UserRole,
+  });
+
+  const proposalsUniqueInstruments: ProposalViewInstrument[] = getUniqueArrayBy(
+    proposalInstruments.flat(),
+    'id'
+  );
+
+  const allSelectedProposalsHaveSameInstruments = [
+    ...proposalInstruments,
+  ].every(
+    (item) =>
+      item?.sort((a, b) => (a && b ? a.id - b.id : 0)).toString() ===
+      proposalInstruments[0]
+        ?.sort((a, b) => (a && b ? a.id - b.id : 0))
+        .toString()
+  );
+
+  const initialValues: Record<string, number | undefined | null> = {};
+
+  proposalsUniqueInstruments.forEach((instrument) => {
+    initialValues[`selectedFapIds_${instrument.id}`] =
+      proposalFapInstruments?.find((i) => i?.instrumentId === instrument.id)
+        ?.fapId || null;
+  });
```

Comment:
Why does this map to `number | undefined | null`? It looks like all the `?.` operators and the `find()` call should return an ID or `undefined`, but those undefined's are caught and replaced with `null`. It feels like one of these options should be safely removable.